src/main/scala/chisel3/stage/phases/AddDedupGroupAnnotations.scala

```diff
@@ -0,0 +1,41 @@
+package chisel3.stage.phases
+
+import firrtl.AnnotationSeq
+import firrtl.options.{Dependency, Phase}
+import firrtl.options.Viewer.view
+import firrtl.transforms.DedupGroupAnnotation
+import scala.annotation.nowarn
+
+import chisel3.experimental.BaseModule
+import chisel3.stage._
+import chisel3.stage.CircuitSerializationAnnotation._
+import chisel3.ChiselException
+import chisel3.experimental.dedupGroup
+import chisel3.internal.firrtl.DefBlackBox
+
+class AddDedupGroupAnnotations extends Phase {
+  override def prerequisites = Seq.empty
+  override def optionalPrerequisites = Seq(Dependency[Convert])
+  override def optionalPrerequisiteOf = Seq.empty
+  override def invalidates(a: Phase) = false
+
+  def transform(annotations: AnnotationSeq): AnnotationSeq = {
+    val chiselOptions = view[ChiselOptions](annotations)
+    val circuit = chiselOptions.chiselCircuit.getOrElse {
+      throw new ChiselException(
+        s"Unable to locate the elaborated circuit, did ${classOf[Elaborate].getName} run correctly"
+      )
+    }
+
+    val skipAnnos = annotations.collect { case x: DedupGroupAnnotation => x.target }
+
+    @nowarn("msg=class Port")
+    val annos = circuit.components.filterNot {
+      case x @ DefBlackBox(id, _, _, _, _) => id.toString contains "Instance"
```

Comment:
What is this accomplishing? We need to find a better way to do this--if we need to expose some information via a new API on `BaseBlackBox`, that's okay.

Reply:
this catches instances of `DefBlackBox` created here: https://github.com/chipsalliance/chisel/blob/e6c07b3760aad1b7ad1c11c9feecc2549ed406df/core/src/main/scala/chisel3/experimental/hierarchy/core/Instance.scala#L142

Reply:
dunno seems like a fairly fast way to accomplish this? there's usually a lot fewer black boxes this will get matched on compared to the total number of components

Reply:
But why do we need to catch those instances?